internal/ingress/controller/parser/parser.go

```diff
@@ -555,7 +555,7 @@ func overrideUpstream(upstream *Upstream,
 func (p *Parser) getUpstreams(serviceMap map[string]Service) ([]Upstream, error) {
 	var upstreams []Upstream
 	for _, service := range serviceMap {
-		upstreamName := service.Backend.ServiceName + "." + service.Namespace + ".svc"
+		upstreamName := service.Backend.ServiceName + "." + service.Namespace + "." + service.Backend.ServicePort.String() + ".svc"
```

Comment:
What happens here when a port is named and is referenced by name in one rule, while referenced by port number in other?
I suspect this will create two different upstreams, which is incorrect.

Could you verify if that's the case or not?

Reply:
It does, but I don't think we can easily avoid it. The `ServicePort` here isn't a proper `ServicePort` object and we cannot access its `port` value. https://github.com/kubernetes/kubernetes/blob/master/pkg/apis/networking/types.go#L357-L364 simply provides the value itself, presumably directly from the Ingress manifest.

Is there harm beyond superfluous configuration? The targets within still end up correct and traffic is routed normally. I don't think there should be many cases where you'd definitely need both to use the same upstream (healthcheck stuff, though I expect most users rely on K8S healthchecks instead), but for those that do, worst case users just have to make sure they use names or numbers consistently.